crates/subspace-runtime/src/object_mapping.rs

```diff
@@ -88,14 +89,26 @@ pub(crate) fn extract_call_block_object_mapping<I: Iterator<Item = Hash>>(
     // Add enum variant to the base offset.
     base_offset += 1;
 
-    if let RuntimeCall::Utility(call) = call {
-        extract_utility_block_object_mapping(
+    match call {
+        // Extract the actual object mappings.
+        RuntimeCall::System(frame_system::Call::remark { remark }) => {
+            if let Some(_call_hash) = successful_calls.next() {
+                objects.push(BlockObject::V0 {
+                    hash: crypto::blake3_hash(remark),
+                    offset: base_offset,
+                });
+            }
```

Comment:
It is not correct to advance `successful_calls` here. `successfull_calls` comes from `validated_object_call_hashes` runtime API that is designed to return hashes of calls that were validated successfully. At https://github.com/autonomys/subspace/pull/2158/files#diff-fbb221bbf0e3e406038048d471e781c3ea4fb288187395151b7d46c6da088a96L924 you can see that it is only applicable to pallets that do have data validation in them and nothing else. It was used to avoid generating object mappings for feed uploads that paid for storage, but failed validation (remember that pallet-feeds did data validation and not every upload necessarily correctly extended the feed forward).

```suggestion
            objects.push(BlockObject::V0 {
                hash: crypto::blake3_hash(remark),
                offset: base_offset,
            });
```